src/pool/pool_scalable.c

```diff
@@ -191,6 +192,13 @@ static void *tbb_calloc(void *pool, size_t num, size_t size) {
         return NULL;
     }
 
+    // synchronizes with release in tbb_free:
+    // if another thread modified memory allocated from tbb and called free
+    // and we end up getting this memory from tbb_malloc above we need to
+    // tell asan that the memset below does not conflict with any preceding
```

Comment:
tell TSAN? still after reading this I don't understand what is the scenario that reproduces this issue. Also, why we don't add such annotation to tbb_malloc?

Reply:
Consider the following case:
```
thread 1                          |  thread 2
ptr = alloc(size) // ptr == 0xAAA |
memset(ptr, 0, size)              |          
free(ptr);                        | 
                                  | ptr = alloc(size) // ptr == 0xAAA
                                  | memset(ptr, 1, size);
                                  | free(ptr);
```

TSAN will report that there is a race on 'ptr' because it thinks that allocation and memset in thread 2 can happen before free(ptr) is called in thread 1 (it doesn't know that there's a happens-before relation between free and alloc, this is what I'm telling it using those annotations).

TBB actually supports TSAN but only if you pass appropriate flag during compilation. So the alternative to this PR could be to just use custom-compiled version of TBB but then you'd need to enforce this in CMake somehow (which does not sound like a good idea).

Reply:
Ok I got it, thanks for the comprehensive explanation.